addons/l10n_es_edi_sii/models/account_edi_format.py

```diff
@@ -566,12 +581,14 @@ def _l10n_es_edi_call_web_service_sign(self, invoices, info_list):
                     inv = candidates
 
             resp_line_state = respl.EstadoRegistro
+            respl_dict = dict(respl)
             if resp_line_state in ('Correcto', 'AceptadoConErrores'):
                 inv.l10n_es_edi_csv = l10n_es_edi_csv
                 results[inv] = {'success': True}
                 if resp_line_state == 'AceptadoConErrores':
                     inv.message_post(body=_("This was accepted with errors: ") + html_escape(respl.DescripcionErrorRegistro))
-            elif respl.RegistroDuplicado:
+            elif (respl_dict.get('RegistroDuplicado') and respl.RegistroDuplicado.EstadoRegistro == 'Correcta'
+                  or cancel and respl_dict.get('CodigoErrorRegistro') == 3001):
```

Comment:
IMO this would be much clearer but not blocking
```suggestion
            elif (
                (respl_dict.get('RegistroDuplicado') and respl.RegistroDuplicado.EstadoRegistro == 'Correcta')
                or
                (cancel and respl_dict.get('CodigoErrorRegistro') == 3001)
            ):
```